src/averageAges.js

```diff
@@ -55,7 +67,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child =>
+    people.find(mother => mother.name === child.mother));
+  const sons = people.filter(child =>
+    people.find(mother => mother.name === child.mother) && child.sex === 'm');
```

Comment:
use children array for sons

```suggestion
  const children = people.filter(child =>
    people.find(mother => mother.name === child.mother));
    
  const sons = people.filter(child =>
    people.find(mother => mother.name === child.mother) && child.sex === 'm');
```